aztec/src/main/kotlin/org/wordpress/aztec/AztecText.kt

```diff
@@ -362,20 +383,88 @@ class AztecText : EditText, TextWatcher {
             removeLeadingStyle(text, LeadingMarginSpan::class.java)
         }
 
-        history.handleHistory(this)
-
         blockFormatter.handleBlockStyling(text, textChangedEventDetails)
         inlineFormatter.handleInlineStyling(textChangedEventDetails)
         lineBlockFormatter.handleLineBlockStyling(textChangedEventDetails)
 
         if (textChangedEventDetails.count > 0 && text.isEmpty()) {
             onSelectionChanged(0, 0)
         }
+
+        // preserve the attributes on the previous list item when adding a new one
+        realignAttributesWhenAddingItem(text)
+
+        history.handleHistory(this)
+
+    }
+
+    private fun realignAttributesWhenAddingItem(text: Editable) {
+        if (textChangedEventDetails.isNewLine()) {
+            val lists = text.getSpans(textChangedEventDetails.inputStart, textChangedEventDetails.inputStart, AztecListSpan::class.java)
+            if (lists.isNotEmpty()) {
+                val listEnd = text.getSpanEnd(lists[0])
+//
```

Comment:
Hmm, I see. I had that working but the fix for the extra line breaks you submitted didn't work in my branch. So I was trying to figure out what was wrong until I finally discovered I disabled [span shortening](https://github.com/wordpress-mobile/WordPress-Aztec-Android/blob/develop/aztec/src/main/kotlin/org/wordpress/aztec/AztecParser.kt#L46) for lists and reverted it. And that broke the emty item at the end, since it removes the last `\n`.